components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/TemplateManagerImpl.java

```diff
@@ -254,34 +246,16 @@ private void validatePaginationParameters(Integer limit, Integer offset) throws
     @Override
     public Template getTemplateById(String templateId) throws TemplateManagementException {
 
-        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
-        try {
-            Resource resource = configManager.getTenantResourceById(templateId);
-            Template template = new ResourceToTemplate().apply(resource);
-            if (resource.getFiles().size() == 1) {
-                InputStream templateScriptInputStream = configManager.getFileById(resource.getResourceType(),
-                        resource.getResourceName(), resource.getFiles().get(0).getId());
-                template.setTemplateScript(IOUtils.toString(templateScriptInputStream));
-            } else {
-                if (log.isDebugEnabled()) {
-                    log.debug(template.getTemplateType().toString() + " can have only one templated object. But the " +
-                            "template with id: " + templateId + " has " + resource.getFiles().size() + " templated " +
-                            "object/s. Therefore templated object is not retrieved.");
-                }
-            }
-            return template;
-        } catch (ConfigurationManagementException e) {
-            if (ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_ID_DOES_NOT_EXISTS.getCode().equals(e
-                    .getErrorCode())) {
-                throw handleClientException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_TEMPLATE_NOT_FOUND, e,
-                        templateId, getTenantDomainFromCarbonContext());
+        List<ReadOnlyTemplateHandler> readOnlyTemplateHandlers =
+                TemplateManagerDataHolder.getInstance().getReadOnlyTemplateHandlers();
+        for (ReadOnlyTemplateHandler readOnlyTemplateHandler: readOnlyTemplateHandlers) {
```

Comment:
can readOnlyTemplateHandlers be null here? if so add a null check

Reply:
addressed in feace15